src/components/StatesButton/StatesButton.spec.tsx

```diff
@@ -62,6 +56,118 @@ describe('StatesButton', () => {
     });
   });
 
+  describe('states', () => {
+    let driver: StatesButtonDriver;
+
+    describe('in progress', () => {
+      beforeEach(() => {
+        onClickSpy = jest.fn();
+        driver = createDriver(
+          <StatesButton
+            {...defaultProps}
+            state={BUTTON_STATES.IN_PROGRESS}
+            onClick={onClickSpy}
+          />,
+        );
+      });
+
+      it('should render in progress content', async () => {
+        expect(await driver.getButtonTextContent()).toEqual('loading');
+      });
+
+      it('should not invoke on click', async () => {
+        await driver.click();
+        expect(onClickSpy).not.toHaveBeenCalled();
+      });
+
+      it('should be disabled', async () => {
+        expect(await driver.isButtonDisabled()).toEqual(true);
+      });
+
+      it('should have aria-busy attribute and not aria-live', async () => {
+        expect((await driver.element()).getAttribute('aria-busy')).toEqual(
+          'true',
+        );
+        expect((await driver.element()).getAttribute('aria-live')).toEqual(
+            'assertive',
+        );
+      });
+    });
+
+    describe('failure', () => {
+      beforeEach(() => {
+        onClickSpy = jest.fn();
+        driver = createDriver(
+          <StatesButton
+            {...defaultProps}
+            state={BUTTON_STATES.FAILURE}
+            onClick={onClickSpy}
+          />,
+        );
+      });
+
+      it('should render failure content', async () => {
+        expect(await driver.getButtonTextContent()).toEqual('failure');
+      });
+
+      it('should invoke on click', async () => {
```

Comment:
Oh, so I think that in your case (Stores' case), you should use the `idle` state for your error, and have the components' error state behave like the success state.
Meaning that the failure state shouldn't be clickable.